snapshots/snapshot_reporter/src/snapshot_reporter.py

```diff
@@ -0,0 +1,135 @@
+
+This lambda should be triggered by a daily CloudWatch event.
+"""
+
+import boto3
+from elasticsearch import Elasticsearch
+import httpx
+import humanize
+
+from dateutil import parser
+import json
+import os
+
+
+def get_secret(secret_id):
+    secrets_client = boto3.client("secretsmanager")
+
+    resp = secrets_client.get_secret_value(SecretId=secret_id)
+
+    try:
+        # The secret response may be a JSON string of the form
+        # {"username": "…", "password": "…", "endpoint": "…"}
+        secret = json.loads(resp["SecretString"])
+    except ValueError:
+        secret = resp["SecretString"]
+
+    return secret
+
+
+def get_elastic_client(elastic_secret_id):
+    secret = get_secret(elastic_secret_id)
+
+    return Elasticsearch(
+        secret["endpoint"], http_auth=(secret["username"], secret["password"])
+    )
+
+
+def get_snapshots(es_client, elastic_index):
+    response = es_client.search(
+        index=elastic_index,
+        body={
+            "query": {
+                "bool": {
+                    "filter": [
+                        {"range": {"snapshotJob.requestedAt": {"gte": f"now-1d/d"}}}
+                    ]
+                }
+            },
+            "sort": [{"snapshotJob.requestedAt": {"order": "desc"}}],
+        },
+    )
+
+    return [hit["_source"] for hit in response["hits"]["hits"]]
+
+
+def prepare_slack_payload(snapshots):
+    def _snapshot_message(snapshot):
+        index_name = snapshot["snapshotResult"]["indexName"]
+        document_count = snapshot["snapshotResult"]["documentCount"]
+        s3_size = snapshot["snapshotResult"]["s3Size"]["bytes"]
+
+        requested_at = parser.parse(snapshot["snapshotJob"]["requestedAt"]).strftime(
+            "%A, %B %-d, %I:%M %p"
+        )
+
+        started_at = parser.parse(snapshot["snapshotResult"]["startedAt"])
+        finished_at = parser.parse(snapshot["snapshotResult"]["finishedAt"])
+
+        time_took = finished_at - started_at
+
+        return "".join(
+            [
+                f"The last snapshot was of index {index_name} at {requested_at}. ",
+                f"It is {humanize.naturalsize(s3_size)}, took {humanize.naturaldelta(time_took)} "
+                f"and contains {humanize.intcomma(document_count)} documents.",
+            ]
+        )
+
+    def _create_header_block(text):
+        return [{"type": "header", "text": {"type": "plain_text", "text": text}}]
+
+    def _create_section_block(text):
+        return [{"type": "section", "text": {"type": "mrkdwn", "text": text}}]
+
+    if snapshots:
+        snapshot = snapshots[0]
+
+        header_block = _create_header_block(":white_check_mark: Catalogue Snapshot")
+        section_block = _create_section_block(_snapshot_message(snapshot))
+    else:
+        header_block = _create_header_block(
+            ":interrobang: Catalogue Snapshot not found"
+        )
+        section_block = _create_section_block("No snapshot found within the last day.")
```

Comment:
I would include a link to the snapshot generator logs here so it's really easy to go debug, but it's not a big deal.